ryelle/css-audit
# Support config file for arguments

This PR contains two main additions: 
* Adds the ability to store audit arguments in a configuration file named css-audit.config.js in the root of the project (a follow up PR can contain smarter config storage using cosmiconfig, for example) - fixes #8 
* Supports multiple property-values audits from the config

A few notes: 
1. The configuration format is slightly different than what is outlined in #8 to work more smoothly with the format that comes from the CLI, but this can be updated if required. 
2. Multiple property-values audits are only supported from configuration, I did not add support for the CLI (though possible if we need this)

A couple other changes to note:
* Refactor the index.js and audit running into a function so that it can be tested
* Remove destructuring of `results` from the JSON format and output the whole result object (this was easier to test and has no other impact as far as I can tell)
* Support for formatting nested directories in the prettier globbing

To test:

Pull down this branch and update the css-audit.config.js to experiment with different audit configurations. Run the command `npm run css-audit -- path/to/css` and confirm that the results display according to what is in the config. 

Make sure the CLI overrides options added in the config e.g. `npm run css-audit -- path/to/css --format=cli-table`. Should display the results as a CLI table.

Still to do:
- [x] Add audits we need from the Google doc
- [x] Use cosmiconfig for smarter config file locating
